numba/tests/test_builtins.py

```diff
@@ -948,6 +959,23 @@ def test_sum_exceptions(self):
 
         self.assertIn('No implementation', str(raises.exception))
 
+    def test_isinstance(self):
+        pyfunc = isinstance_usecase 
+        cfunc = jit(nopython=True)(pyfunc)
+
+        inputs = (
+            3,        # int
+            5.0,      # float
+            "Hello",  # string
+            1j,       # complex
+        )
+
+        for inpt in inputs:
+            expected = pyfunc(inpt)
+            got = cfunc(inpt)
+            self.assertEqual(expected, got)
+
```

Comment:
Would be good to add a load more tests here, for example:
```python
from numba import njit
import numpy as np

@njit
def foo(x):
    if isinstance(x, (int, int)):
        print('is int')
    else:
        print('is not int')

print(foo(100))
```
this currently prints "is not int", which is a bug?!

This is entirely illegal and yet still compiles and runs:
```python
from numba import njit
import numpy as np

@njit
def foo(x):
    if isinstance(x, ('foo',)):
        print('is int')
    else:
        print('is not int')

print(foo(100))
```
and maybe not for this PR, but it'd be good to consider how to get things like:
```python
from numba import njit
import numpy as np

@njit
def foo(x):
    if isinstance(x, tuple):
        print('A')
    else:
        print('B')

foo(100)
```
to work.

Reply:
I've fixed the first two cases. In the last case, what is `tuple`?

Reply:
> I've fixed the first two cases. In the last case, what is `tuple`?

Thanks, feel free to try more variations! `tuple` here is literally the `tuple` builtin function:
```pycon
In [9]: tuple?
Init signature: tuple(iterable=(), /)
Docstring:     
Built-in immutable sequence.
```

Reply:
Oh, got it. You mean, any container like `list`, `dict`, `set`, `tuple`, ...